File: FakeBet/Models/Match.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using FakeBet.Extensions;

namespace FakeBet.Models
{
    public enum MatchStatus
    {
        NonStarted,
        OnGoing,
        Ended
    }


    public class Match
    {
        // teamAName+teamBName+Category+dateTime.Ticks
        [Key] public string MatchId { get; set; }

        public string Category { get; set; }

        public DateTime MatchTime { get; set; }

        public MatchStatus? Status { get; set; }

        public string TeamAName { get; set; }

        public int TeamAPoints { get; set; }

        public string TeamBName { get; set; }

        public int TeamBPoints { get; set; }

        public float PointsRatio { get; set; }

        public IEnumerable<Vote> Votes { get; set; }

        public void GenerateDefaultValues()
        {
            MatchId = TeamAName.RemoveAllSpaces() + TeamBName.RemoveAllSpaces() +
                      Category +
                      MatchTime.Ticks;
            Status = DetermineMatchStatus(MatchTime);
            TeamAPoints = 0;
            TeamBPoints = 0;
            PointsRatio = 1;
            Votes = new List<Vote>();
        }

        private static MatchStatus DetermineMatchStatus(DateTime date)
        {
            return DateTime.Compare(date, DateTime.Now) > 0 ? MatchStatus.NonStarted : MatchStatus.Ended;
        }
    }
}

File: FakeBet.Site/src/app/models/match.ts
export class Match {
  matchId: string;
  category: string;
  matchTime: Date;
  status: MatchStatus;
  teamAName: string;
  teamAPoints: number;
  teamANationalityCode: string;
  teamBName: string;
  teamBPoints: number;
  teamBNationalityCode: string;
  votes: any[]; // todo change to some vote interface or model
}

export enum MatchStatus {
  NonStarted = 0,
  OnGoing = 1,
  Ended = 2
}
